CMakeLists.txt

```diff
@@ -143,29 +126,56 @@ else() # OSS RediSearch
 
     set_target_properties(redisearch PROPERTIES LINKER_LANGUAGE CXX)
     setup_shared_object_target(redisearch "")
-    target_link_libraries(redisearch redisearch-geometry VectorSimilarity ${REDISEARCH_LIBS} ${CMAKE_LD_LIBS})
+    target_link_libraries(redisearch
+            redisearch-geometry
+            VectorSimilarity
+            redisearch-coord
+            ${LIBUV_LIBS}
+            ${HIREDIS_LIBS}
+            ${SSL_LIBS}
+            ${REDISEARCH_LIBS}
+            ${CMAKE_LD_LIBS})
 
     extract_debug_symbols(redisearch)
 
     add_dependencies(redisearch VectorSimilarity)
+
+endif()
+
+if (${COORD_TYPE} STREQUAL "oss")
+    target_compile_definitions(redisearch-coord PRIVATE RS_CLUSTER_OSS)
+elseif (${COORD_TYPE} STREQUAL "rlec")
+    target_compile_definitions(redisearch-coord PRIVATE RS_CLUSTER_ENTERPRISE)
+    # TODO: why calling it module and not redisearch???
+    set_target_properties(${PROJECT_NAME} PROPERTIES OUTPUT_NAME "module-enterprise")
+else()
+    message(FATAL_ERROR "Invalid TYPE (='${TYPE}'). Should be either 'oss' or 'rlec'")
 endif()
 
 #----------------------------------------------------------------------------------------------
 
-if (BUILD_COORDINATOR)
-    set(BUILD_SEARCH_UNIT_TESTS OFF)
-endif()
+if (BUILD_SEARCH_UNIT_TESTS)
+    enable_testing()
+
+    add_subdirectory(tests/cpptests/redismock)
+
+    add_library(testdeps STATIC ${FINAL_OBJECTS} $<TARGET_OBJECTS:redismock>)
+
+    target_link_libraries(testdeps
+            ${LIBUV_LIBS}
+            redisearch-geometry
+            redismock
+            VectorSimilarity
+            ${HIREDIS_LIBS}
+            ${SSL_LIBS})
```

Comment:
can you explain why we need to link VSS, libuv, hiredis, libssl to the unit tests?

Reply:
Good point, I checked and actually we don't. I removed `testdeps` completely and used only `redismock` instead